Tella/Scenes/Uwazi/ViewModel/UwaziTemplateViewModel.swift

```diff
@@ -0,0 +1,165 @@
+//
+//  UwaziTemplateViewModel.swift
+//  Tella
+//
+//  Created by Gustavo on 31/07/2023.
+//  Copyright © 2023 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import Combine
+
+class UwaziTemplateViewModel: ObservableObject {
+    
+    var mainAppModel : MainAppModel
+    
+    @Published private var templates : [CollectedTemplate] = []
+    @Published private var downloadedTemplates : [CollectedTemplate] = []
+    
+    @Published var templateItemsViewModel : [TemplateItemViewModel] = []
+    @Published var templateCardsViewModel : [TemplateCardViewModel] = []
+    
+    @Published var isLoading: Bool = false
+    @Published var serverName : String
+    var subscribers = Set<AnyCancellable>()
+    var server: Server
+    
+    var tellaData: TellaData? {
+        return self.mainAppModel.vaultManager.tellaData
+    }
+    
+    init(mainAppModel : MainAppModel, server: Server) {
+        
+        self.mainAppModel = mainAppModel
+        self.server = server
+        self.serverName = server.name ?? ""
+    }
+    
+    func getTemplates() {
+        self.isLoading = true
+        Task {
+            guard let id = self.server.id else { return }
+            guard let locale = self.tellaData?.getUwaziLocale(serverId: id) else { return }
+            let template = try await UwaziServerRepository().handleTemplate(server: self.server, locale: locale)
+            template.receive(on: DispatchQueue.main).sink { completion in
+                self.handleGetTemplateCompletion(completion)
+            } receiveValue: { templates in
+                self.handleRecieveValue(self.mapToCollectedTemplate(serverId: id, templates))
+            }.store(in: &subscribers)
+        }
+    }
+    
+    
+    fileprivate func mapToCollectedTemplate(serverId: Int, _ templates: [UwaziTemplateRow]) -> [CollectedTemplate] {
+        return templates.map { template in
+            return CollectedTemplate(serverId: serverId,
+                                     templateId: template.id,
+                                     serverName: self.server.name ?? "",
+                                     username: self.server.username,
+                                     entityRow: template,
+                                     isDownloaded: false,
+                                     isFavorite: false,
+                                     isUpdated: false)
+        }
+    }
+    
+    func downloadTemplate(template: CollectedTemplate) {
+        var template = template
+        Toast.displayToast(message: "“\(template.entityRow?.translatedName ?? "")” “\(LocalizableUwazi.uwaziAddTemplateSavedToast.localized)”")
```

Comment:
I think Toast is a view and it's better to display it in the view not in the view model